src/Uno.UWP/Devices/Geolocation/Geolocator.Android.cs

```diff
@@ -1,186 +1,346 @@
+		}
+
+
+		public static async Task<GeolocationAccessStatus> RequestAccessAsync()
+		{
+			var status = GeolocationAccessStatus.Allowed;
+
+			if (!await PermissionsHelper.CheckFineLocationPermission(CancellationToken.None))
+			{
+				status = await PermissionsHelper.TryGetFineLocationPermission(CancellationToken.None)
+					? GeolocationAccessStatus.Allowed
+					: GeolocationAccessStatus.Denied;
+
+				BroadcastStatus(PositionStatus.Initializing);
+				if (status == GeolocationAccessStatus.Allowed)
+				{
+					BroadcastStatus(PositionStatus.Ready);
+
+					// If geolocators subscribed to PositionChanged before the location permission was granted,
+					// make sure to initialize these geolocators now so they can start broadcasting.
+					foreach (var subscriber in _positionChangedSubscriptions)
+					{
+						subscriber.Key.TryInitialize();
+					}
+				}
+				else
+				{
+					BroadcastStatus(PositionStatus.Disabled);
+
+					foreach (var subscriber in _positionChangedSubscriptions)
+					{
+						subscriber.Key.WaitForPermissionFromBackground();
+					}
+				}
+			}
+
+			return status;
+		}
+
+		public void WaitForPermissionFromBackground()
+		{
+			CoreApplication.Resuming -= CoreApplication_Resuming;
+			CoreApplication.Resuming += CoreApplication_Resuming;
+		}
+
+		private void CoreApplication_Resuming(object sender, object e)
+		{
+			CoreDispatcher.Main.RunAsync(
+				priority: CoreDispatcherPriority.Normal,
+				handler: InitializeIfPermissionIsGranted
+			);
+		}
+
+		private async void InitializeIfPermissionIsGranted()
+		{
+			// If the user has granted the location permission while the app was in background, Initialize
+			if (await PermissionsHelper.CheckFineLocationPermission(CancellationToken.None))
+			{
+				TryInitialize();
+				CoreApplication.Resuming -= CoreApplication_Resuming;
+			}
+		}
+
+		private LocationManager InitializeLocationProvider(double desiredAccuracy)
+		{
+			var locationManager = (LocationManager)global::Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+
+			var criteriaForLocationService = new Criteria
+			{
+				Accuracy = Accuracy.Coarse
+			};
+
+			var acceptableLocationProviders = locationManager.GetProviders(criteriaForLocationService, true);
+
+			if (acceptableLocationProviders.Any())
+			{
+				_locationProvider = acceptableLocationProviders.First();
+			}
+			else
+			{
+				_locationProvider = String.Empty;
+			}
+
+			return locationManager;
+		}
+
+		partial void StartPositionChanged()
+		{
+			_positionChangedSubscriptions.TryAdd(this, 0);
+		}
+
+		partial void StopPositionChanged()
+		{
+			_positionChangedSubscriptions.TryRemove(this, out var _);
+		}
+
+		public async Task<Windows.Devices.Geolocation.Geoposition> GetGeopositionAsync(TimeSpan maximumAge, TimeSpan timeout)
+		{
+
+			_locationManager = (LocationManager)Android.App.Application.Context.GetSystemService(Android.Content.Context.LocationService);
+
+			_reportInterval = 1000;

```

Comment:
This seems arbitrary. Why 1s ?

Reply:
Doc says, "The default report interval is 1 second, or as frequent as the hardware can support". So, I use 1 second, skipping second part of this sentence, becase under Android, geolocation takes too much power (using shorter report interval can drain device accumulator in several minutes).